### DISCUSSION TITLE
Cannot login successfully in specific situation.

### DISCUSSION THREAD

**Discussion by tedhsieh17** [pixiv/vroid-sdk-developers#108] [Q&A]
Hello, I have a question about the login. The user cannot log in successfully if our app is recycled by the system when opening the external browser by invoking the SDK login function. I would like to know how to handle this situation. Thanks.

Here is a video to simulate the scenario by turning on the `Don't keep activities` from the `Developer options` and the test app is from the example project attached to the VRoid SDK `(Assets/VRoidSDK/Examples/CharacterModelExample/Scenes/Example.unity)`

https://github.com/pixiv/vroid-sdk-developers/assets/77952903/25a69c17-10bd-4b07-84b9-5ba1c7bd77c9

**Comment by nyamadan** [comment]
Hi. I’ll investigate the issue.

**Reply by tedhsieh17** [reply]
Hi @nyamadan , thanks for the replay, and would like to know if there is any good news.

**Reply by nyamadan** [reply]
We have reproduced this issue.
The activity seems to be killed when the browser is opened.

We plan to fix this.

**Reply by tedhsieh17** [reply]
Looking forward to the fixed version being released and would like to know if there is an estimated time for the new library. Thanks a lot.

**Reply by tedhsieh17** [reply]
Hi @nyamadan , is this issue fixed or planned to be fixed at which version?

**Reply by nyamadan** [reply]
We tried to fix it, but we couldn't.

"Don't keep activities" is an emulation function for events that occur when memory is low.
We do not believe this problem occurs that often.

Could you please tell me when exactly this is a problem?

**Reply by tedhsieh17** [reply]
On some devices (ex: Pixel 3a XL), we often encounter the issue that the system recycles our app during the login process. Is there any suggestion to workaround it?

**Reply by tedhsieh17** [reply]
Hi @nyamadan , do you have any suggestion or workaround to make it login successfully? We tried to stop other apps except for our app and the browser but no luck.

**Reply by nyamadan** [reply]
In [v0.5.0](https://github.com/pixiv/vroid-sdk-developers/discussions/132), the app link implementation was changed to ChromeCustomTabs.
[This makes it less likely that the app will be killed.](https://developer.chrome.com/docs/android/custom-tabs#:~:text=Apps%20launching%20a%20Custom%20Tab%20won%27t%20be%20evicted%20by%20the%20system%20during%20the%20Tabs%20use%20%2D%20its%20importance%20is%20raised%20to%20the%20%22foreground%22%20level.)

However, if "Don't keep activities" is enabled, the program is forced to be killed...

**Reply by tedhsieh17** [reply]
Hi @nyamadan, we updated the SDK to a version higher than `v0.5.0,` but some users still report they cannot login successfully. It seems the system recycles and restarts our app between receiving and sending deep links to our app. Do you have any suggestion or workaround to avoid that and make it login successfully? 🙏

**Comment by gao80840** [comment]
In conclusion, there doesn't seem to be a solution.

It appears that "Activity" can be killed when memory usage becomes high. We have made adjustments to make it less likely to be killed, but this does not guarantee a resolution.

Also, it seems that Auth0.android has given up on making it compatible with the "Don't keep activities" setting.

The discussion at the URL below is helpful.
https://github.com/auth0/Auth0.Android/issues/242

Since it's a specification on the OS side and there's little we can do, unfortunately, the solution will likely involve asking users to stop background apps and take other measures.
